### PR TITLE
SN-736   NetForge - Details of Navigation Reference

### PR DESCRIPTION
**Pull request by mchernikov80**
- Apply DisplayDetails flag value changes in EntityService.MapNavigation() method;
- Add EditDetails flag in the NavigationMetadata and apply this to navigation options and metadata DTO methods; add appropriate fluent API SetAsEditable() method;
- Add customization for navigation reference, to go to navigation reference edit page directly from parent entity list view page…n() method;
- Change the \docs\NAVIGATIONS.md

### CODE REVIEW

File: demo/Saritasa.NetForge.Demo/Infrastructure/Admin/ShopAdminConfiguration.cs [hunk 1]
```diff
@@ -65,7 +64,8 @@ public void Configure(EntityOptionsBuilder<Shop> entityOptionsBuilder)
                             .SetIsSortable(true)
                             .SetSearchType(SearchType.ContainsCaseInsensitive);
                     })
-                    .SetDisplayDetails(true);
```
Comment (AlexanderMakarov124) [inline demo/Saritasa.NetForge.Demo/Infrastructure/Admin/ShopAdminConfiguration.cs:68]: I think we don't need this method anymore

File: src/Saritasa.NetForge.Blazor/Controls/EntityPropertyColumns.razor.cs [hunk 2]
```diff
@@ -185,6 +189,17 @@ private async Task OpenDialogAsync(object navigationInstance, NavigationMetadata
...
 
+    /// <summary>
+    /// Edit navigation entity by its identifier.
+    /// </summary>
+    private async Task NavigateToEditing(NavigationMetadataDto navigationMetadata, object navigationIdentifier)
+    {
+        var entityMetadata = await EntityService.GetEntityByTypeAsync(navigationMetadata.ClrType!, CancellationToken.None);
+
+        NavigationService.NavigateTo<Mvvm.ViewModels.EditEntity.EditEntityViewModel>(
+            parameters: new[] { entityMetadata.PluralName, navigationIdentifier });
```
Comment (AlexanderMakarov124) [inline src/Saritasa.NetForge.Blazor/Controls/EntityPropertyColumns.razor.cs:200]: Use collection expressions

**Review by AlexanderMakarov124 — CHANGES_REQUESTED** [review]

File: demo/Saritasa.NetForge.Demo/Infrastructure/Admin/ShopAdminConfiguration.cs (hunk 1 above)
Comment (mchernikov80) [reply demo/Saritasa.NetForge.Demo/Infrastructure/Admin/ShopAdminConfiguration.cs:68]: I'm not sure... But i think we need this to assign DisplayDetails flag in the navigation options. This is necessary to create navigation OpenDialog window link  with navigation entity details.
I tested it

File: docs/NAVIGATIONS.md [hunk 3]
```diff
@@ -84,4 +84,30 @@ public void Configure(EntityOptionsBuilder<Shop> entityOptionsBuilder)
...
+    entityOptionsBuilder
+        .IncludeNavigation<Address>(shop => shop.Address, navigationOptionsBuilder =>
+        {
+            navigationOptionsBuilder
+                .IncludeProperty(address => address.Id, builder =>
+                {
+                    builder.SetDisplayName("Address Id");
+                })
+                .SetIsDisplayDetails(true)
+				.SetAsEditable();
```
Comment (AlexanderMakarov124) [inline docs/NAVIGATIONS.md:110]: Replace tabs with spaces here

**Review by AlexanderMakarov124 — APPROVED** [review]

**Pull request merged**